omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,240 @@
+    public static final String REMOTE_IDENTIFIER_SOURCE_TYPE = "RemoteIdentifierSource";
+    public static final String IDENTIFIER_POOL_SOURCE_TYPE = "IdentifierPool";
+
+    private IdentifierSourceService service;
+
+    @Before
+    public void setUp() throws Exception {
+        executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+        this.service = Context.getService(IdentifierSourceService.class);
+    }
+
+    @Override
+    public String getURI() {
+        return "idgen/identifiersource";
+    }
+
+    @Override
+    public String getUuid() {
+        return SEQUENTIAL_IDENTIFIER_SOURCE_UUID;
+    }
+
+    @Override
+    public long getAllCount()  {
+        return service.getAllIdentifierSources(false).size();
+    }
+
+    @Override
+    @Test(expected = ResourceDoesNotSupportOperationException.class)
+    public void shouldGetAll() throws Exception {
+        super.shouldGetAll();
+    }
+
+    @Test
+    public void shouldGetAnIdentifierSourceByUuid() throws Exception {
+
+        SimpleObject result = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+        assertNotNull(result);
+        assertEquals(getUuid(), PropertyUtils.getProperty(result, "uuid"));
+
+    }
+
+    @Test
+    public void shouldEditASequentialIdentifierSource() throws Exception {
+
+        String name = "{\"name\":\"Updated Sequential Identifier Name\"}";
+        assertEquals("Test Sequential Generator", service.getIdentifierSourceByUuid(getUuid()).getName());
+        handle(newPostRequest(getURI() + "/" + getUuid(), name));
+        assertEquals("Updated Sequential Identifier Name", service.getIdentifierSourceByUuid(getUuid()).getName());
+
+    }
+
+    @Test
+    public void shouldEditARemoteIdentifierSource() throws Exception {
+
+        String name = "{\"name\":\"Updated Remote Identifier Name\"}";
+        assertEquals("Test Remote Source", service.getIdentifierSourceByUuid(REMOTE_IDENTIFIER_SOURCE_UUID).getName());
+        handle(newPostRequest(getURI() + "/" + getUuid(), name));
+        assertEquals("Updated Remote Identifier Name", service.getIdentifierSourceByUuid(getUuid()).getName());
+
+    }
+
+    @Test
+    public void shouldEditAnIdentifierPoolSource() throws Exception {
+
+        String name = "{\"name\":\"Updated Identifier Pool Description\"}";
+        assertEquals("Test Identifier Pool", service.getIdentifierSourceByUuid(POOL_SOURCE_UUID).getName());
+        handle(newPostRequest(getURI() + "/" + getUuid(), name));
+        assertEquals("Updated Identifier Pool Description", service.getIdentifierSourceByUuid(getUuid()).getName());
+
+    }
+    
+    @Test
+    public void shouldSaveASequentialIdentifierGenerator() throws Exception {
+
+        long initialIdentifierSourceCount = getAllCount();
+
+        SimpleObject identifierSource = new SimpleObject();
+        identifierSource.add("name", "test sequential identifier");
```

Comment:
Why don't you make the description different from name?

Reply:
fixed